### ISSUE TITLE
How to properly restore?

### ISSUE THREAD

**Issue by tiotrom** [YunoHost/issues#1743]
I have a ynh install on IP X. I backup everything, including  the ynh settings and all that. I upgraded my server but kept the same IP. Backups are on my local computer. I reinstalled ynh on the new server. Copied the  ynh backups to the ynh backups folder. I then go to ynh admin panel, backups, and I see them all there. I clicked to restore all settings. And now nothing works :). 

I thought that I can restore the ynh settings, then the apps, on any new server. As easy as explained above.

**Comment by alexAubin** [comment]
... Well do you have something more accurate and specific to share than "nothing works" ...

**Comment by tiotrom** [comment]
Sure I mean what to share? It restored. Then I got this error "The server closed the connection instead of answering it. Has nginx or the yunohost-api been restarted or stopped for some reason? (Error code/message: 0 error)"

Then rebooted the server and I cannot access anything.

**Comment by tiotrom** [comment]
It restored the following:
```

/etc/ldap/ldap.conf | conf/ldap/ldap.conf
-- | --
/etc/ldap/slapd.ldif | conf/ldap/slapd.ldif
/etc/ssh | conf/ssh
/etc/ssowat | conf/ssowat
/etc/yunohost/firewall.yml | conf/ynh/firewall/firewall.yml
/etc/yunohost/certs | conf/ynh/certs
/var/mail | data/mail
/etc/metronome | conf/xmpp/etc
/var/lib/metronome | conf/xmpp/var
/etc/nginx/conf.d | conf/nginx
/etc/cron.d/yunohost-diagnosis | conf/cron/yunohost-diagnosis
/etc/cron.d/yunohost-dyndns | conf/cron/yunohost-dyndns
/etc/yunohost/current_host | conf/ynh/current_host
/usr/share/yunohost/hooks/restore/08-conf_ssh | hooks/restore/08-conf_ssh
/usr/share/yunohost/hooks/restore/20-conf_ynh_firewall | hooks/restore/20-conf_ynh_firewall
/usr/share/yunohost/hooks/restore/23-data_mail | hooks/restore/23-data_mail
/usr/share/yunohost/hooks/restore/32-conf_cron | hooks/restore/32-conf_cron
/usr/share/yunohost/hooks/restore/21-conf_ynh_certs | hooks/restore/21-conf_ynh_certs
/usr/share/yunohost/hooks/restore/26-conf_xmpp | hooks/restore/26-conf_xmpp
/usr/share/yunohost/hooks/restore/40-conf_ynh_currenthost | hooks/restore/40-conf_ynh_currenthost
/usr/share/yunohost/hooks/restore/29-conf_nginx | hooks/restore/29-conf_nginx
/usr/share/yunohost/hooks/restore/14-conf_ssowat | hooks/restore/14-conf_ssowat
/usr/share/yunohost/hooks/restore/05-conf_ldap | hooks/restore/05-conf_ldap
/home/yunohost.backup/tmp/ynh_core_backup/backup.csv | backup.csv
/home/yunohost.backup/tmp/ynh_core_backup/info.json | info.json
/home/yunohost.backup/tmp/ynh_core_backup/conf | conf
/home/yunohost.backup/tmp/ynh_core_backup/data | data
  |  
  |  

```

**Comment by alexAubin** [comment]
Then let's start with the basic info ...

What kind of hardware are you running (I don't know what 'IP X' is) .. Is it computer / ARM board / an online VPS

**Comment by tiotrom** [comment]
Online VPS. Sorry, IP X meant to say is a particular IP, "X". Here's more output:

```
admin@trom:~$ systemctl status nginx
● nginx.service - A high performance web server and a reverse proxy server
   Loaded: loaded (/lib/systemd/system/nginx.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Sat 2021-02-20 17:45:15 CET; 24s ago
     Docs: man:nginx(8)
  Process: 10683 ExecStartPre=/usr/sbin/nginx -t -q -g daemon on; master_process on; (code=exited, status=1/FAILURE)
```
Maybe nginx stopped working after I installed YNH=?

**Comment by alexAubin** [comment]
Yes ... maybe ... So apparently "nothing works" but you have a way to run commands on the server, so at least quite a lot of things works (it's not like the server "did not boot at all")

So yes, for some reason nginx seems broken ... So let's run `nginx -t` to check it's happy with the configuration, and naively `systemctl restart nginx` and see if that fixes the issue, who knows

**Comment by tiotrom** [comment]
Sorry when I said nothing works I meant YNH does not work.

```
sudo nginx -t
nginx: [emerg] open() "/etc/nginx/conf.d/ds.conf" failed (2: No such file or directory) in /etc/nginx/nginx.conf:61
nginx: configuration file /etc/nginx/nginx.conf test failed

```

```
systemctl restart nginx
Failed to restart nginx.service: Access denied
See system logs and 'systemctl status nginx.service' for details.
```

```
sudo systemctl restart nginx
Job for nginx.service failed because the control process exited with error code.
See "systemctl status nginx.service" and "journalctl -xe" for details.
```

**Comment by tiotrom** [comment]
ds.conf is there, but is a symlink. That's how it backed-up...that's how it restored.

**Comment by alexAubin** [comment]
A symlink pointing to what ? 

```
ls -l /etc/nginx/conf.d/ds.conf
```

**Comment by tiotrom** [comment]
/etc/nginx/conf.d/ds.conf -> /etc/onlyoffice/documentserver/nginx/ds.conf

hm....

**Comment by tiotrom** [comment]
We had onlyoffice installed is true, but I just restored the core ynh settings for now...is this breaking it?

**Comment by alexAubin** [comment]
Hmyeah that's what I was thinking about, onlyoffice does it stupid stuff and savagely injects its nginx conf and it's a huge mess ...

What about the restauration of onlyoffice, did it work ? Was it restored at all ?

**Comment by tiotrom** [comment]
Ok let me restore onlyoffice to see how it goes. Interesting, had 0 clues about this :P

**Comment by tiotrom** [comment]
oh WOW that's it, it works now. Jizas haha. So this is a big deal then!? You should add a warning somewhere about it. Or is it?

**Comment by tiotrom** [comment]
btw I got again that database error that I fixed with this https://github.com/YunoHost/issues/issues/1731 - maybe is worth adding this to the troubleshooting guide? I could add them if you tell me how. Idk...

**Comment by alexAubin** [comment]
>oh WOW that's it, it works now. Jizas haha. So this is a big deal then!? You should add a warning somewhere about it. Or is it?

Yes it's clearly a bug, though it's kind of a combinaition of two edge-cases or weird behaviors : a) the fact that we naively backup/restore the entire nginx conf folder, which we shouldnt have to, because the regenconf process is supposed to regen the conf entirely just like it should be, and b) the fact that onlyoffice savagely injects a bunch of configuration from its .deb and doesn't let you (well "us", the yunohost packaging team) handle the nginx conf

So in a perfect world, imho the real action to take is to change the backup/restore process such that it doesnt backup/restore the entire nginx folder ... which is in fact something we're discussing for a few years now, but are yet to actually change

**Comment by alexAubin** [comment]
>btw I got again that database error that I fixed with this #1731 - maybe is worth adding this to the troubleshooting guide? I could add them if you tell me how. Idk...

Hmpf yeah we can add some documentation about it I suppose ... though personally I'm always feeling "meh" because documenting dozens of workaround is not an actual fix ... Funnily-enough, a similar issue started to appear in our recent RPi image and I don't quite understand what the fuck is happening suddenly and why only for RPi ... Mysql really is fucking mysterious ...

**Comment by tiotrom** [comment]
Oh I see yes. So sure troubleshooting is useful, but ideally it should not happen. Thanks again for all the help and input. I learned a bit from you, that's for sure.
